### PR TITLE
Remake GCE for Foreman Google plugin

### PR DESCRIPTION
**Pull request by Lennonka**
GCE compute resource provider was split from Foreman core into the standalone [foreman_google](https://github.com/theforeman/foreman_google) plugin with new installer arguments and changes in the compute resource data and UIs.

* [x] I am familiar with the [contributing](https://github.com/theforeman/foreman-documentation/blob/master/CONTRIBUTING.md) guidelines.

Please cherry-pick my commits into:

* [x] Foreman 3.5/Katello 4.7 (planned Satellite 6.13)
* [ ] Foreman 3.4/Katello 4.6 (EL8 only)
* [ ] Foreman 3.3/Katello 4.5 on EL7 & EL8 (Satellite 6.12 on EL8 only)
* [ ] Foreman 3.2/Katello 4.4 on EL7 & EL8
* [ ] Foreman 3.1/Katello 4.3 on EL7 & EL8 (Satellite 6.11 EL7/8, orcharhino 6.1 on EL7, orcharhino 6.2 on EL7/8)
* For Foreman 3.0 or older, please create a separate PR.
* We do not accept PRs for Foreman 2.3 or older.

### CODE REVIEW

**Comment by ekohl** [comment]
> With reimplementation in Ruby 3

This is not true. Initially it was suggested that [fog-google](https://github.com/fog/fog-google) was a blocker to the Ruby 3 migration, but that gem has since been updated to support Ruby 3.

File: guides/common/modules/proc_adding-a-google-gce-connection.adoc [hunk 1]
```diff
@@ -5,31 +5,50 @@ Use this procedure to add Google Compute Engine (GCE) as a compute resource in {
...
++
+[options="nowrap", subs="+quotes,verbatim,attributes"]
+----
+# chmod 0600 /usr/share/foreman/_gce_key_.json
+----
+. On {ProjectServer}, restore SELinux context for your service account key:
++
+[options="nowrap", subs="+quotes,verbatim,attributes"]
+----
+# restorecon -vv /usr/share/foreman/_gce_key_.json
```
Comment (ekohl) [inline guides/common/modules/proc_adding-a-google-gce-connection.adoc:43]: Users should not write files to `/usr/share/foreman`. This feels like configuration, which belongs in `/etc`.

Second: it's better to make it owned by `root:foreman` and use mode `0640`. That way the `foreman` user can only read it, and that's what we want.

**Review by ekohl — COMMENTED** [review]
I know you're just moving things around, but this is the first time I review it and I see some pre-existing problems.

About the key: it really surprises me we read a file from disk for it. That really goes against the design we typically have where we try to avoid that. Mostly because if you do end up load balancing then it's better to read it from the database. It also looks like it ends up being stored in the DB in the password (if I read https://github.com/theforeman/foreman_google/blob/c6c2c1ed6cb42f1f9a7f160746ec6139d7f61d2c/app/controllers/foreman_google/api/v2/compute_resources_extensions.rb#L15-L18 correctly), which implies you can actually remove the file from disk after importing.

@stejskalleos shouldn't we make a way where the content of the JSON file is sent to the API, rather than the file path?

File: guides/common/modules/proc_uninstalling-google-gce-plugin.adoc [hunk 2]
```diff
@@ -8,7 +8,8 @@ If you have previously installed the Google GCE plugin but don't use it anymore
 +
 [options="nowrap", subs="+quotes,verbatim,attributes"]
 ----
-# {package-remove-project} foreman-gce
-# {foreman-installer} --no-enable-foreman-compute-gce
+# {installer-scenario} \
```
[the remark this answers is not in the document]
Comment (stejskalleos) [reply guides/common/modules/proc_uninstalling-google-gce-plugin.adoc:11]: Yop, this section should be remove, also confirmed by Eric in https://bugzilla.redhat.com/show_bug.cgi?id=2163798#c2

**Review by stejskalleos — COMMENTED** [review]
> shouldn't we make a way where the content of the JSON file is sent to the API, rather than the file path?

We probably should do it the way you say, but it was never implemented as we wanted to have plugin same behavior as core feature before.

File: guides/common/modules/proc_adding-a-google-gce-connection.adoc (hunk 1 above)
Comment (Lennonka) [reply guides/common/modules/proc_adding-a-google-gce-connection.adoc:43]: If users download it and they can remove it afterwards, perhaps root's home directory would be better?
Config in `/etc` feels more permanent.

Comment (ekohl) [reply guides/common/modules/proc_adding-a-google-gce-connection.adoc:43]: The home directory would not be readable by the Foreman user; there does appear to be a weird API where a file path on the server is given and where the user `foreman` will read it. That's why I asked @stejskalleos if we can send the contents over the API. That would actually work in a networked environment, which Hammer is supposed to work in.

File: guides/common/modules/proc_uninstalling-google-gce-plugin.adoc (hunk 2 above)
[the remark this answers is not in the document]
Comment (Lennonka) [reply guides/common/modules/proc_uninstalling-google-gce-plugin.adoc:11]: Removed for upstream as well, I gather.

**Review by stejskalleos — APPROVED** [review]
:green_apple: LGTM

**Review by ekohl — COMMENTED** [review]
Minor comments, otherwise :+1:

File: guides/common/modules/con_provisioning-cloud-instances-on-google-compute-engine.adoc [hunk 3]
```diff
@@ -6,8 +6,11 @@
 :CRname: Google Compute Engine
 
 {ProjectName} can interact with Google Compute Engine (GCE), including creating new virtual machines and controlling their power management states.
+
 ifdef::satellite[]
 You can only use golden images supported by {Team} with {Project} for creating GCE hosts.
+
+The GCE compute resource provider is enabled by default in {Project}.
```
[the remark this answers is not in the document]
Comment (Lennonka) [reply guides/common/modules/con_provisioning-cloud-instances-on-google-compute-engine.adoc:13]: I'm hesitant to include the word "always".

[the remark this answers is not in the document]
Comment (Lennonka) [reply guides/common/modules/con_provisioning-cloud-instances-on-google-compute-engine.adoc:13]: Perhaps this sentence is unnecessary?

[the remark this answers is not in the document]
Comment (Lennonka) [reply guides/common/modules/con_provisioning-cloud-instances-on-google-compute-engine.adoc]: Compromise :)

[the remark this answers is not in the document]
Comment (ekohl) [reply guides/common/modules/con_provisioning-cloud-instances-on-google-compute-engine.adoc]: I was going to suggest `is included in ...`, but I think it's fair to leave it out. Users should get used to either being told to install it, or assume it's not needed.

**Review by ekohl — APPROVED** [review]

**Comment by adamlazik1** [comment]
Merged and cherry-picked:
   de98f94a1..8219c6e2e  3.5 -> 3.5

**Pull request merged**